l5kit/l5kit/simulation/unroll.py

```diff
@@ -0,0 +1,226 @@
+            agents_update_dict[(scene_idx, next_agent["track_id"])] = np.expand_dims(next_agent, 0)
+        dataset.set_agents(frame_idx, agents_update_dict)
+
+    @staticmethod
+    def update_ego(dataset: SimulationDataset, frame_idx: int, input_dict: Dict[str, torch.Tensor],
+                   output_dict: Dict[str, torch.Tensor]) -> None:
+        """Update ego across scenes for the given frame index.
+
+        :param dataset: The simulation dataset
+        :param frame_idx: index of the frame to modify
+        :param input_dict: the input to the ego model
+        :param output_dict: the output of the ego model
+        :return:
+        """
+        input_dict = {k: v.cpu().numpy() for k, v in input_dict.items()}
+        output_dict = {k: v.cpu().numpy() for k, v in output_dict.items()}
+
+        world_from_agent = input_dict["world_from_agent"]
+        yaw = input_dict["yaw"]
+        pred_trs = transform_points(output_dict["positions"][:, :1], world_from_agent)
+        pred_yaws = yaw + output_dict["yaws"][:, :1]
+
+        dataset.set_ego_for_frame(frame_idx, 0, pred_trs, pred_yaws)
+
+
+from l5kit.configs import load_config_data
+from l5kit.data import ChunkedDataset, LocalDataManager
+from l5kit.dataset import EgoDataset
+from l5kit.rasterization import build_rasterizer
+
+
+class MockModel(torch.nn.Module):
+    def __init__(self):
+        super(MockModel, self).__init__()
+
+    def forward(self, x: Dict[str, torch.Tensor]):
+        centroids = x["centroid"]
+        bs = len(centroids)
+
+        positions = torch.zeros(bs, 12, 2, device=centroids.device)
+        positions[..., 0] = 1
+
+        yaws = torch.zeros(bs, 12, device=centroids.device)
+
+        return {"positions": positions, "yaws": yaws}
+
+
+if __name__ == '__main__':
+    zarr_dt = ChunkedDataset("/tmp/l5kit_data/scenes/sample.zarr").open()
+    print(zarr_dt)
+
+    cfg = load_config_data(
+        "/Users/lucabergamini/Desktop/l5kit/examples/agent_motion_prediction/agent_motion_config.yaml")
+
+    cfg["raster_params"]["ego_center"] = (0.25, 0.5)
+    cfg["raster_params"]["pixel_size"] = (0.5, 0.5)
+    cfg["raster_params"]["raster_size"] = (224, 224)
+
+    cfg["model_params"]["history_num_frames"] = 0
+    cfg["model_params"]["future_num_frames"] = 12
+    cfg["raster_params"]["map_type"] = "py_semantic"
+
+    rast = build_rasterizer(cfg, LocalDataManager("/tmp/l5kit_data"))
+
+    sim_cfg = SimulationConfig(use_ego_gt=False,
+                               use_agents_gt=False,
+                               disable_new_agents=False,
+                               distance_th_far=10,
+                               distance_th_close=30,
+                               start_frame_index=0,
+                               num_simulation_steps=20)
+    sim = SimulationLoop(sim_cfg, MockModel(), MockModel())
+    sim.unroll(EgoDataset(cfg, zarr_dt, rast), [5, 6, 7, 8])
```

Comment:
I think it is better to move this to the tests.

Reply:
same here